### PR TITLE
Update saml-keycloak.md

### PR DESCRIPTION
**Pull request by taqattack**
Revised SAML setup with Keycloak

> [!IMPORTANT]
> If you are merging into master, please add either a `backport` or a `no-backport` label to this PR. You will not be able to merge until you do this step. Refer to the section [Do I need to backport this PR?](https://www.notion.so/metabase/Metabase-Branching-Strategy-6eb577d5f61142aa960a626d6bbdfeb3?pvs=4#89f80d6f17714a0198aeb66c0efd1b71) in the Metabase Branching Strategy document for more details.

> **Warning**
>
> If that is your first contribution to Metabase, please sign the [Contributor License Agreement](https://docs.google.com/a/metabase.com/forms/d/1oV38o7b9ONFSwuzwmERRMi9SYrhYeOrkbmNaq9pOJ_E/viewform) (unless it's a tiny documentation change). Also, if you're attempting to fix a translation issue, please submit your changes to our [POEditor project](https://poeditor.com/join/project/ynjQmwSsGh) instead of opening a PR.

Closes https://github.com/metabase/metabase/issues/[issue_number]

### Description

Describe the overall approach and the problem being solved.

### How to verify

Describe the steps to verify that the changes are working as expected.

1. New question -> Sample Dataset -> ...
2. ...

### Demo

_Upload a demo video or before/after screenshots if sensible or remove the section_

### Checklist

- [ ] Tests have been added/updated to cover changes in this PR

### CODE REVIEW

File: docs/people-and-groups/saml-keycloak.md [hunk 1]
```diff
@@ -16,44 +16,40 @@ For more information, check out our guide for [authenticating with SAML](./authe
...
     - Click **Set Password** to save your changes.
-4. Create a new SSO client from **Manage** > **Clients** > **Create**.
-    - **Client ID**: Enter “metabase” in lowercase.
-    - **Client Protocol**: Select “saml” from the dropdown.
+4. Create a new SSO client from **Manage** > **Clients** > **Create**
+    - **Client ID**: Enter `metabase` in lowercase.
+    - **Client type**: Select `SAML` from the dropdown.
+    - Click **Next**.
+    - **Valid Redirect URIs**: The URL where you are hosting your Metabase instance followed by a slash (/) and an asterisk (*). For example, if you are hosting Metabase locally at `http://localhost:3000`, the URL would be `http://localhost:3000/*`.
+    - **Home URL**: From your Metabase instance, go to **Admin settings** > **Authentication** > **SAML**. Look for the field *URL the IdP should redirect back to* and use it for the Home URL.
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:28]:
```suggestion
    - **Home URL**:  In your Metabase, go to **Admin settings** > **Authentication** > **SAML**. You'll find your Home URL in the field **URL the IdP should redirect back to**.
```

File: docs/people-and-groups/saml-keycloak.md [hunk 2]
```diff
@@ -16,44 +16,40 @@ For more information, check out our guide for [authenticating with SAML](./authe
...
-5. Configure the SSO client from the form that appears after saving:
-    - **Client Signature Required**: DISABLE
-    - **Valid Redirect URIs**: URL where you are hosting your Metabase instance followed by
-     a slash (/) and an asterisk (*). For example, `http://localhost:3000/*`.
-    - **Base URL**: Fill this in with the value under “URL the IdP should redirect back to” from your Metabase **Admin settings** > **Authentication** > **SAML**.
-    - Click **Save**.
-6. Map user data to your SSO client from **Mappers** > **Add Builtin**.
-    - [Mapping attributes from users in Keycloak to Metabase](#mapping-attributes-from-users-in-keycloak-to-metabase)
+
+5. Disable key signing for SSO client (optional). If you would like to enable key signing, please follow [this guide](https://www.metabase.com/docs/latest/people-and-groups/authenticating-with-saml#settings-for-signing-sso-requests-optional).
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:31]:
```suggestion
5. (Optional) Disable key signing for SSO client.  See [settings for signing SSO requests](https://www.metabase.com/docs/latest/people-and-groups/authenticating-with-saml#settings-for-signing-sso-requests-optional).
```

Links should be descriptive.

File: docs/people-and-groups/saml-keycloak.md [hunk 3]
```diff
@@ -16,44 +16,40 @@ For more information, check out our guide for [authenticating with SAML](./authe
...
+    - Click **Keys** tab.
+    - **Client signature required:** Off.
+ 
+6. Map user attributes from Metabase to SSO client. 
+    - Click **Client scopes** tab.
+    - Click `metabase-dedicated`
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:37]:
```suggestion
    - Click `metabase-dedicated`.
```
If one item in a list terminates in punctuation, they all should.

File: docs/people-and-groups/saml-keycloak.md [hunk 4]
```diff
@@ -16,44 +16,40 @@ For more information, check out our guide for [authenticating with SAML](./authe
...
+    - Click **Add predefined mappers**
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:38]:
```suggestion
    - Click **Add predefined mappers**.
```

File: docs/people-and-groups/saml-keycloak.md [hunk 5]
```diff
@@ -16,44 +16,40 @@ For more information, check out our guide for [authenticating with SAML](./authe
...
+    - [Map attributes from users in Keycloak to Metabase](#mapping-attributes-from-users-in-keycloak-to-metabase)
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:39]:
```suggestion
    - [Map attributes from users in Keycloak to Metabase](#mapping-attributes-from-users-in-keycloak-to-metabase).
```

File: docs/people-and-groups/saml-keycloak.md [hunk 6]
```diff
@@ -16,44 +16,40 @@ For more information, check out our guide for [authenticating with SAML](./authe
...
 1. Go to your Metabase **Admin settings** > **Authentication** > **SAML**.
-2. Enter the information from step 8 above:
-    - **SAML Identity Provider URL**: the URL from 8.1.
-    - **SAML Identity Provider Issuer**: the URL from 8.2.
-    - **SAML Identity Provider Certificate**: the string from 8.3. Take care when inserting this string -- the setup won't work if any letters or special characters are wrong!
-    - **SAML Application Name**: metabase
+2. From the XML file from Step 7 above:
+    - **SAML Identity Provider URL**: Insert the URL that appears right after the following string: `Binding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Location=`
+    - **SAML Identity Provider Issuer**: Insert the URL that appears right after `entityID=`.
+    - **SAML Identity Provider Certificate**: Insert the long string that appears after the `<X509Certificate>` tag. Take care when inserting this string -- the setup won't work if any letters or special characters are wrong!
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:51]:
```suggestion
    - **SAML Identity Provider Certificate**: Input the long string that appears after the `<X509Certificate>` tag. Take care when inserting this string: if any letters or special characters are added or off,  the setup won't work.
```

File: docs/people-and-groups/saml-keycloak.md [hunk 7]
```diff
@@ -69,7 +65,7 @@ Let's say we want email, name, and surname to be passed between the client (Meta
     - **SAML Attribute Name**: the name that Metabase expects to receive.
     - **SAML Attribute NameFormat**: select “Basic” from the dropdown menu.
 
-You can find the attribute values from your Metabase **Admin settings** > **Authentication** > **SAML** > **Attributes**.
+You can find/edit the attribute values from your Metabase **Admin settings** > **Authentication** > **SAML** > **Attributes**.
```
Comment (jeff-bruemmer) [inline docs/people-and-groups/saml-keycloak.md:68]:
```suggestion
You can edit the attribute values from your Metabase **Admin settings** > **Authentication** > **SAML** > **Attributes**.
```

**Review by jeff-bruemmer — APPROVED** [review]
Thanks, @taqattack! Approved and left some suggestions.

**Review by dragonsahead — APPROVED** [review]

**Pull request merged**